org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/SbtDependenciesAnalysis.scala

```diff
@@ -0,0 +1,16 @@
+package org.scalaide.core.internal.builder.zinc
+
+import sbt.inc.IncOptions
+import sbt.inc.Analysis
+import org.scalaide.core.internal.builder.EclipseBuildManager
+import org.eclipse.core.resources.IFile
+
+trait SbtDependenciesAnalysis {
+  def latestAnalysis(incOptions: => IncOptions): Analysis
+}
+
+trait CachedSbtDependenciesAnalysis extends SbtDependenciesAnalysis {
+  def analysisStore: IFile
+}
+
+trait SbtAwareBuildManager extends EclipseBuildManager with SbtDependenciesAnalysis
```

Comment:
Ok, after reading the whole code I realize there are still two implementations:

- the new one, using build scopes
- the old one, using one compiler run

In this case I suggest extending `EclipseBuildManager` with the required additional methods from the Sbt implementation, and this one `latestAnalysis`, so that the interface is enough to do all the operations, without pattern matching on the type, as we do know in several places.